src/views/SpecialResolution.vue

```diff
@@ -252,25 +239,30 @@ export default class SpecialResolution extends Mixins(
         this.setResource(this.specialResolutionResource)
 
         // initialize Fee Summary data
-        this.setFilingData(this.specialResolutionResource.filingData)
+        const filingData = [this.specialResolutionResource.filingData]
+        if (this.hasBusinessNameChanged) {
+          filingData.push({
+            filingTypeCode: FilingCodes.SPECIAL_RESOLUTION_NAME_CHANGE,
+            entityType: CorpTypeCd.COOP,
+            priority: false
+          })
+        }
+        filingData.forEach(fd => {
+          fd.futureEffective = this.getEffectiveDateTime.isFutureEffective
+        })
+        this.setFilingData(filingData)
+
+        this.setFilingData(filingData)
```

Comment:
I find this part confusing, so other codes using this array style become difficult to understand too. Correct me if I'm wrong. I guess that you want to add the three key-values `filingTypeCode`, `entityType` and `priority` to the dictionary `filingData`, as well as `futureEffective` in the next line. What I don't understand is why you use an array for this, and as a result, all files using `filingData` need to be changed because of it? It's not very convincing to me that we should turn `filingData` into an array.

Reply:
You need an array because the base filing has an additional fee schedule for a NAME CHANGE.  (You'll see when you change the NR)

Reply:
Also use travis8814@gmail.com for the NR (you'll see what I mean)

Reply:
![image](https://user-images.githubusercontent.com/3484109/183776745-b79794f5-a000-4624-aaf2-8775eae0477f.png)

Reply:
Change of Name and Special resolution are two different fee schedules (which is why we needed the array, most of the other filings are just a single fee schedule (with priority / future included)

Reply:
Here's what it looks like in the pay admin: 

![image](https://user-images.githubusercontent.com/3484109/183777127-0a567976-4e8e-4c71-8803-9bc6f4e65ff6.png)

![image](https://user-images.githubusercontent.com/3484109/183777159-f271568c-c8fa-4894-aca6-5f38b1885266.png)

![image](https://user-images.githubusercontent.com/3484109/183777174-64aed05a-5f0e-4ed0-bf7e-565c1fbdccf7.png)

Reply:
I find it difficult to process this array style. If the backend is not currently returning an array, or the frontend is not sending an array in the payload, I think it's easier to use another getter/setter for this special case. This is working properly, so it should be good. I'll resolve this conversation once this duplicate `this.setFilingData(filingData)` is removed.

Reply:
Arg, sorry about the duplicate I think that was from the merge from main I believe.

Would making another getter/setter for example `nameChangeFilingData` and append that to `filingData` would that make more sense?

Let me know what you have in mind and I'll gladly implement it.